pydci/pydci.py

```diff
@@ -64,19 +64,31 @@ def query(self, q, k, max_retrieve, max_composite_visit):
             d-dimensional query points
         k : int
             Number of neighbors to return
-        max_retrieve : int
-            The maximum number of candidates to pull from each composite index
-        max_composite_visit : int
-            The maximum number of times to visit each composite index
+        max_retrieve_const : float
+            Scaling factor for maximum number of candidates to pull from each
+            composite index. Values in the range of 1-10 work well.
+        max_composite_visit_const : float
+            Scaling factor for the maximum number of times to visit each
+            composite index. Values in the range of 1-10 work well.
 
         Returns
         -------
-        k nearest neighbors
+        k nearest neighbors as indices and points, as well as the number of
+        candidates searched through and number of insertions and deletions in
+        priority queues to create candidate list
 
         """
+        # set parameters via scaling factors
+        # scaling of parameters is based on theory and qualitative uning
+        max_retrieve = int(max_retrieve_const * self._num_simple *
+                           k * np.log(self._num_points / k) ** 2)
+        max_candidates = int(max_candidates_const * self._num_simple * k *
+                             np.log(self._num_points / k) ** 2)
```

Comment:
We don't have access to the step parameter in pydci, but in your tests, you added step onto each of these? Is that very important? I ran the code in your test notebook using this version, and it still seemed to look good.

Reply:
No, only reason to have it there was for the first iteration really. Could just add k or small multiple of k